### PR TITLE
Adding link comparison script.

### PR DESCRIPTION
**Pull request by Dli3**
[MWPW-146942](https://jira.corp.adobe.com/browse/MWPW-146942)
Adding link comparison script.

This script will go in and grab the links on page A and B. Then check that page B has the same hrefs associated to the link texts found in page A.

### CODE REVIEW

File: tests/bacom-blog/link-comparison.test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import fs from 'fs';
+import { test } from '@playwright/test';
+import { findLinks, checkForDiff } from '../../utils/linkComparison.js';
+
+const data = JSON.parse(fs.readFileSync('./data/bacom-blog/stagedBlogUrls.json', 'utf8'));
```
Comment (JackySun9) [inline tests/bacom-blog/link-comparison.test.js:5]: you don't include the json file here

File: tests/bacom-blog/link-comparison.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        if (res.status() === 404) {
+          results[urlB] = 404;
+          test.skip();
+        }
+
+        await page.waitForLoadState('domcontentloaded');
+        linksB = await findLinks(page);
+      });
+
+      await test.step('3. Verify that the links are the same in url A and B', async () => {
```
Comment (JackySun9) [inline tests/bacom-blog/link-comparison.test.js:55]: how do you make sure, urlA is comparing urlB, I mean if there is more links in urlB, the links will not match

**Review by JackySun9 — CHANGES_REQUESTED** [review]

File: tests/bacom-blog/link-comparison.test.js (hunk 2 above)
Comment (Dli3) [reply tests/bacom-blog/link-comparison.test.js:55]: @JackySun9 We just want to make sure the links from A were not shuffled in B. There could've been authoring changes to B at the time of running the script, so we're not checking if they have the same length.

File: tests/bacom-blog/link-comparison.test.js (hunk 1 above)
Comment (Dli3) [reply tests/bacom-blog/link-comparison.test.js:5]: Do we want to include data files in our repo? @JackySun9 
cc @skumar09 @amauch-adobe

Comment (JackySun9) [reply tests/bacom-blog/link-comparison.test.js:5]: are the urls dynamic?

File: tests/bacom-blog/link-comparison.test.js (hunk 2 above)
Comment (JackySun9) [reply tests/bacom-blog/link-comparison.test.js:55]: could you give some examples?

File: utils/linkComparison.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  Object.entries(a).forEach(([key, value]) => {
+    const aValue = value ? value.toString() : value;
+    const bValue = b[key] ? b[key].toString() : b[key];
+    if (aValue !== bValue) {
+      diff[key] = [aValue, bValue];
+    }
+  });
+
+  return diff;
```
Comment (JasonHowellSlavin) [inline utils/linkComparison.js:44]: Could use reduce, no need to change. 

```suggestion
export async function checkForDiff(a, b) {
  if (Object.keys(b).length === 0) return {};

  return Object.entries(a).reduce((rdx, [key, value]) => {
    const aValue = value ? value.toString() : value;
    const bValue = b[key] ? b[key].toString() : b[key];
    if (aValue !== bValue) {
      rdx[key] = [aValue, bValue];
    }
    return rdx;
  }, {});
```

**Review by JasonHowellSlavin — COMMENTED** [review]
Looks good, nothing needs to change, but some suggestions.

File: tests/bacom-blog/link-comparison.test.js (hunk 2 above)
Comment (amauch-adobe) [reply tests/bacom-blog/link-comparison.test.js:55]: @JackySun9 the reason for this test is to address a CSO issue where between bulk updating blog files the urls were no longer the same as they were before, meaning the links swapped around in the file that caused the links to no longer be valid.  It has led to discussions about the Franklin process of the word doc in sharepoint being converted to MarkDown for Franklin to use and then converted back to a Word doc with the updates in Sharepoint being there being the culprit for the scrambling of the links (according to Microsoft support).  This test is to help that situation.

File: utils/linkComparison.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/**
+ * @description Finds the main tag and returns all the a tags found.
```
Comment (amauch-adobe) [inline utils/linkComparison.js:2]: might want to change the 'a tags' to '<a> tags' in this comment so it is more clear.

**Review by amauch-adobe — APPROVED** [review]
Approved, made a small nit pick grammar change in a comment, but otherwise approved.

File: utils/linkComparison.js (hunk 3 above)
Comment (Dli3) [reply utils/linkComparison.js]: Updated. Thanks @JasonHowellSlavin .

File: tests/bacom-blog/link-comparison.test.js (hunk 1 above)
Comment (Dli3) [reply tests/bacom-blog/link-comparison.test.js:5]: I didn't plan for it to be ran in CI/CD. @amauch-adobe 

@JackySun9 I've added the json file.

File: tests/bacom-blog/link-comparison.test.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        const res = await page.goto(urlB);
+
+        if (res.status() === 404) {
+          results[urlB] = 404;
+          test.skip();
+        }
+
+        await page.waitForLoadState('domcontentloaded');
+        linksB = await findLinks(page);
+      });
```
[the remark this answers is not in the document]
Comment (Dli3) [reply tests/bacom-blog/link-comparison.test.js]: Thanks for the suggestion. I've updated it. @JasonHowellSlavin

**Review by skumar09 — APPROVED** [review]
@Dli3 : This looks good, as discussed will raise enhancement pr for link difference reporting, (like csv, or html)

**Review by JackySun9 — APPROVED** [review]

**Pull request merged**